assets/ts/schedule/components/line-diagram/LineDiagram.tsx

```diff
@@ -58,23 +57,40 @@ const LineDiagram = ({
   const liveData = useRealtime(route, directionId, true);
   const [query, setQuery] = useState("");
 
+  const dispatch = useDispatch();
+  const currentState = useSelector((state: StoreProps) => state);
+
   const allStops: RouteStop[] = stopTree
     ? stopIds(stopTree).map(stopId => stopForId(stopTree, stopId))
     : routeStopList;
   const filteredStops: RouteStop[] = allStops.filter(stop =>
     stop.name.toLowerCase().includes(query.toLowerCase())
   );
 
+  const changeOrigin = (origin: SelectedOrigin): void => {
+    dispatch({
+      type: "CHANGE_ORIGIN",
+      newStoreValues: {
+        selectedOrigin: origin
+      }
+    });
+    // reopen modal depending on choice:
+    dispatch({
+      type: "OPEN_MODAL",
+      newStoreValues: {
+        modalMode: origin ? "schedule" : "origin"
```

Comment:
**suggestion**: I _think_ if you added a test, or adjusted a current test, to use the "origin" case here, that'll bump up your coverage to 95%. so close!

Reply:
I am wondering why it is under 95% as it was meeting the coverage requirement when I was running it locally

Reply:
![Screenshot 2024-06-06 at 9 16 57 AM](https://github.com/mbta/dotcom/assets/2679229/9f9a1e07-9148-4bcf-8697-908d2ee3061e)
I changed zero tests, and it isn't complaining about coverage locally